### ISSUE TITLE
Rel attribute: the UI doesn't display the actual set values

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#15915]
Noticed while testing #10128 and it applies to all the links "Rel" attribute fields.

For example, testing on the Image block > Sidebar > Link Settings and playing with "Open in New Tab" and "Link Rel".

Tested steps:

Click "Open in New Tab": noreferrer and noopener values are added to the "Link Rel" field:

<img width="322" alt="Screenshot 2019-05-30 at 10 28 11" src="https://user-images.githubusercontent.com/1682452/58623736-f235d980-82ce-11e9-8977-b6485a0a2f71.png">

OR: enter a custom value:

<img width="316" alt="Screenshot 2019-05-30 at 10 29 13" src="https://user-images.githubusercontent.com/1682452/58623777-05e14000-82cf-11e9-9d5d-c29093c4557f.png">

In both cases the values displayed in the UI and the ones rendered in the front end match. So far so good.

### Enter a custom value _and then_ click "Open in New Tab":
- save
- the UI displays only the custom value
- values on the front end are correct: `myrel noopener noreferrer`

<img width="311" alt="custom + open in new tab" src="https://user-images.githubusercontent.com/1682452/58624358-50af8780-82d0-11e9-9c17-7b540ce5b7ff.png">

Refresh the page: displayed values are now correct: `myrel noopener noreferrer`

### Click "Open in New Tab" to disable it:
- the UI still displays all the three values `myrel noopener noreferrer`

<img width="298" alt="Screenshot 2019-05-30 at 11 59 04" src="https://user-images.githubusercontent.com/1682452/58625293-7473cd00-82d2-11e9-8475-0813fffe232a.png">

### Reset everything, restart from scratch: click "Open in New Tab" _and then_ add a custom value:
- save
- the UI displays all the three values and values in the front end are correct 

<img width="315" alt="open in new tab + custom" src="https://user-images.githubusercontent.com/1682452/58625482-d92f2780-82d2-11e9-848c-479c61345bb7.png">

### Click "Open in New Tab" to disable it:
- save
- the UI still displays all the three values `noreferrer noopener mynewrel`
- values in the front end are still `noreferrer noopener mynewrel`

<img width="312" alt="remove open in new tab" src="https://user-images.githubusercontent.com/1682452/58626106-43949780-82d4-11e9-9ccb-c98d3a5992b7.png">

### Recap
There are probably more flows that produce incorrect / mismatched values. Trying to summarize:
- The UI may display incorrect values.
- Values in the front end may be incorrect, especially in combination with the target attribute value. For example there are cases where `target="_blank"` is not rendered but `rel="noreferrer noopener myrel"` is, which is incorrect
- In some cases an empty `rel=""` is rendered, which is a bit pointless

### Thoughts
- I'm not sure the Rel field should display the `noreferrer noopener` values to start with
- From a UI perspective, I'd say "Open in New Tab" and the Rel field should be close each other. Having the CSS class field in between doesn't help users understand these two fields are related.

**Comment by kjellr** [comment]
The mismatch should definitely be fixed! That's really confusing. 

In terms of the other recommendations: 

> I'm not sure the Rel field should display the noreferrer noopener values to start with

In general, if an attribute value is active, we should display it. If it's not active, we should hide it. I'm a little unclear of the recommendation here — can you explain the reasoning? 

> From a UI perspective, I'd say "Open in New Tab" and the Rel field should be close each other. Having the CSS class field in between doesn't help users understand these two fields are related.

This makes sense. Let's swap the CSS class and Link Rel positions to get this working. 👍 

Thank you!

**Comment by afercia** [comment]
> > I'm not sure the Rel field should display the noreferrer noopener values to start with

> In general, if an attribute value is active, we should display it. If it's not active, we should hide it. I'm a little unclear of the recommendation here — can you explain the reasoning?

`noreferrer noopener` are values related to security. I'd tend to think users shouldn't mess with them. In the same way, WordPress parses the post content and automatically adds `noreferrer noopener` to links with `target="_blank"` used within the post content. As far as I know, it doesn't inform users or displays these values in any way and for good reasons :)

**Comment by kjellr** [comment]
Ah cool. Thanks for the explanation. Yeah, I think it may make sense to just hide those at all times then.

**Comment by donmhico** [comment]
@afercia @kjellr If you have the time, can you test my PR #17398? Thank you!
